app/src/lib/stores/repositories-store.ts

```diff
@@ -531,33 +546,132 @@ export class RepositoriesStore extends BaseStore {
+      dbID
+    )
+
+    if (branchProtectionsFound === undefined) {
+      return this.loadAndCacheBranchProtection(dbID)
+    }
+
+    return branchProtectionsFound
+  }
+
+  private async isBranchProtected(
+    dbID: number,
     branchName: string
+  ): Promise<boolean> {
+    const key = getKey(dbID, branchName)
+
+    const existing = this.protectionEnabledForBranchCache.get(key)
+    if (existing === true) {
+      return existing
+    }
+
+    const result = await this.db.protectedBranches.get([dbID, branchName])
+
+    // if no row found, this means no protection is found for the branch
+    const value = result !== undefined
+
+    this.protectionEnabledForBranchCache.set(key, value)
+
+    return value
+  }
+
+  /**
+   * Check if any branch protection settings are enabled for the repository
+   * through the GitHub API.
+   *
+   * Will also check the parent repository if is defined and the fork repository
+   * does not have this branch protected.
+   */
+  public async hasBranchProtectionsConfigured(
+    gitHubRepository: GitHubRepository
   ): Promise<boolean> {
     if (gitHubRepository.dbID === null) {
       return fatalError(
         'unable to get protected branches, GitHub repository has a null dbID'
       )
     }
 
-    const key = getKey(gitHubRepository.dbID, branchName)
+    const { dbID } = gitHubRepository
+    const isBranchProtectionsFound = await this.findOrCacheBranchProtections(
+      dbID
+    )
+
+    if (isBranchProtectionsFound) {
+      return true
+    }
 
-    const existing = this.branchProtectionCache.get(key)
-    if (existing !== undefined) {
-      return existing
+    if (
+      gitHubRepository.parent !== null &&
+      gitHubRepository.parent.dbID !== null
+    ) {
+      const parentID = gitHubRepository.parent.dbID
+      return await this.findOrCacheBranchProtections(parentID)
     }
 
-    const result = await this.db.protectedBranches.get([
-      gitHubRepository.dbID,
-      branchName,
-    ])
+    return false
+  }
 
-    const value = result !== undefined
+  /**
+   * Check if the given branch for the repository is protected through the
+   * GitHub API.
+   *
+   * Will also check the parent repository if is defined and the fork repository
+   * does not have this branch protected.
+   */
+  public async isBranchProtectedOnRemote(
+    gitHubRepository: GitHubRepository,
+    branchName: string
+  ): Promise<boolean> {
+    if (gitHubRepository.dbID === null) {
+      return fatalError(
+        'unable to get protected branches, GitHub repository has a null dbID'
+      )
+    }
 
-    this.branchProtectionCache.set(key, value)
+    const { dbID } = gitHubRepository
+    const isProtected = await this.isBranchProtected(dbID, branchName)
 
-    return value
+    if (isProtected) {
+      return true
+    }
+
+    if (
+      gitHubRepository.parent !== null &&
+      gitHubRepository.parent.dbID !== null
+    ) {
+      const parentID = gitHubRepository.parent.dbID
```

Comment:
And I'm going to skip this case here because we don't actually fetch branch protections for the upstream branch, so unless you already have this populated because it's also tracked in the app, this is not going to be present in the database.

And this feels like it's not focused on the workflow we're interested in - **commits to a protected branch on the current remote**. If we start trying to be clever here by being aware of the upstream repository I think it's going to distort the numbers here.